aten/src/ATen/DLConvertor.cpp

```diff
@@ -101,8 +101,9 @@ static DLDevice getDLDevice(const Tensor& tensor, const int64_t& device_id) {
       ctx.device_type = DLDeviceType::kDLROCM;
       break;
     case DeviceType::XPU:
-      ctx = at::detail::getXPUHooks().getDLPackDeviceFromATenDevice(
-          ctx, tensor.device(), tensor.data_ptr());
+      ctx.device_type = DLDeviceType::kDLOneAPI;
+      ctx.device_id =
+          at::detail::getXPUHooks().getGlobalIdFromDevice(tensor.device());
```

Comment:
Why do we have to override `ctx.device_id` here?

Reply:
for dlpack, we use the device index enumerated from sycl::device::get_devices(), aka global id, as the parameter. So we need to convert the device index of the framework into a global id here.